### PR TITLE
48 cleanup raw array accesses in annotate ast

### LINKED ISSUE #48: Cleanup raw array accesses in annotateAst
# Description

The earlier expressions still use raw array accessing instead of functions from astHelper.ts.

# Sub-tasks

- [x] Replace all array accesses with astHelper calls

### PR DESCRIPTION
**Pull request by HectorPeeters**
# Description

Give a short description of what was done in this merge request.

Closes #48 

# Changes

1. Nested array accesses have been removed from annotateAst.ts
2. Move the insertAttribute function to astHelper.ts

# Deletes Source Branch?

Yes:    Work is done, the branch can be safely deleted.

# How Many Approvals?

* 2 Approvals (Normal features and bugs)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -111,6 +111,8 @@ function compareSpecificity(xpathStringA: string, xpathStringB: string): -1 | 0
 	return parseXPath(xpathStringA).specificity.compareTo(parseXPath(xpathStringB).specificity);
 }
 
+evaluateXPath('1 - 1');
```
Comment (maanpalle) [inline src/index.ts]: Do not touch index.ts!

**Review by maanpalle — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (HectorPeeters) [reply src/index.ts]: Fixed it!

**Review by maanpalle — APPROVED** [review]

File: src/parsing/astHelper.ts [hunk 2]
```diff
@@ -195,6 +195,23 @@ function getQName(ast: IAST): QName {
...
+function insertAttribute(ast: IAST, name: string, data: any) {
+	if (typeof ast[1] === 'object' && !Array.isArray(ast[1])) {
+		ast[1][name] = data;
+	} else {
+		const obj = {};
+		obj[name] = data;
+		ast.splice(1, 0, obj);
+	}
+}
+
```
Comment (yuanzexiong) [inline src/parsing/astHelper.ts:214]: Well yes, this function indeed belongs here.

**Review by yuanzexiong — APPROVED** [review]
Cool changes well done!

File: src/parsing/astHelper.ts (hunk 2 above)
Comment (HectorPeeters) [reply src/parsing/astHelper.ts:214]: Yeah that's what I thought as well

**Review by MTimmerije — APPROVED** [review]
Seems fine to me, should make everything easier to read.

**Pull request merged**